libkbfs/delayed_cancellation_test.go

```diff
@@ -102,6 +102,25 @@ func TestDelayedCancellationCleanupWhileNotEnabled(t *testing.T) {
 	}
 }
 
+func TestDelayedCancellationSecondEnable(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := makeContextWithDelayedCancellation(t)
+	defer cancel()
+
+	err := EnableDelayedCancellationWithGracePeriod(ctx, 0)
+	if err != nil {
+		t.Fatalf("1st EnableDelayedCancellationWithGracePeriod failed: %v", err)
+	}
+	time.Sleep(5 * time.Millisecond) // make sure async ops sorts out
+	// parent context is not canceled; second "enable" should succeed even it's
```

Comment:
You are right. That's what this PR fixes. Before this PR, grace period started in `EnableDelayedCancellationWithGracePeriod`. So the second time would fail since it's after the grace period. This PR makes sure the grace period only happens after parent context is canceled. So if it's never canceled, the grace period never starts, so 2nd call to `EnableDelayedCancellationWithGracePeriod` should still succeed. This added test makes sure after grace period and calling `EnableDelayedCancellationWithGracePeriod` for the 2nd time, it still succeeds as long as the parent context is not canceled.